### PR TITLE
Add integration test for HAProxy, Memcached, Nginx, Java/JMX 

### PR DESCRIPTION
**Pull request by bjrara**
**Why do we need it?**
This is a follow-up PR of https://github.com/aws-observability/aws-otel-test-framework/pull/251.
In this PR, we add more test cases for CloudWatch container insight demo workloads.

### CODE REVIEW

**Comment by pingleig** [comment]
@bjrara can you resolve the conflict since I merged #251

**Comment by bjrara** [comment]
> @bjrara can you resolve the conflict since I merged #251

Done. Thanks

File: terraform/containerinsight_eks_prometheus/jmx/jmx.tf [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    template {
+      metadata {
+        labels = {
+          app = "tomcat-bad-traffic-generator"
+        }
+      }
+      spec {
+        container {
+          name              = "tomcat-traffic-generator"
+          image             = "ellerbrock/alpine-bash-curl-ssl"
```
Comment (pingleig) [inline terraform/containerinsight_eks_prometheus/jmx/jmx.tf]: I think we can make the image and the common part of the command a var or local

File: validator/src/main/resources/expected-data-template/eksContainerInsightExpectedMetrics.mustache [hunk 2]
```diff
@@ -184,3 +184,464 @@
       value: 2
     - name: envoy_http_conn_manager_prefix
       value: ingress
+
```
Comment (pingleig) [inline validator/src/main/resources/expected-data-template/eksContainerInsightExpectedMetrics.mustache:187]: So one test run contains all workload and expect all the metrics (memcache, jmx, nginx etc.) to show up?

Comment (bjrara) [reply validator/src/main/resources/expected-data-template/eksContainerInsightExpectedMetrics.mustache:187]: Yes

**Review by pingleig — APPROVED** [review]
LGTM

**Pull request merged**